### PR TITLE
[SOLR-16490] Create a v2 equivalent for /admin/cores BACKUPCORE+RESTORECORE

### PR DESCRIPTION
**Pull request by iamsanjay**
https://issues.apache.org/jira/browse/SOLR-16490

# Description

The /admin/cores?action=BACKUPCORE API have no equivalent in the v2 paradigm. It's an internal API used by Collection Handler to create of backup of collection.

# Solution

Current v1 API: GET /solr/admin/cores?action=BACKUPCORE&core=coreName
Proposed v2 API: POST /api/cores/backups/backupName {...}

More info on V2 api mapping is in Google Sheet available [here](https://docs.google.com/spreadsheets/d/1HAoBBFPpSiT8mJmgNZKkZAPwfCfPvlc08m5jz3fQBpA/edit?usp=sharing), which includes additional information about each API, such as the hosting RequestHandler.

# Tests

Added few test cases. This PR is in draft mode. Later on, more test cases will be added to this commit.

# Checklist

Please review the following and check all that apply:

- [x] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [x] I have created a Jira issue and added the issue ID to my pull request title.
- [ ] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [x] I have developed this patch against the `main` branch.
- [ ] I have run `./gradlew check`.
- [x] I have added tests for my changes.
- [ ] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

### CODE REVIEW

**Comment by gerlowskija** [comment]
Hi @iamsanjay , sorry for the delay in getting to this.  I've been mostly attending to a blocker for the JAX-RS and v2 API work, SOLR-16531.  But with that recently cleared away I'm turning back to these API conversions.  Hopefully the delay hasn't put you off of collaborating here!

Going to test your change out locally today and hopefully we can understand why the response format doesn't look quite right.

**Review by gerlowskija — CHANGES_REQUESTED** [review]
Hi @iamsanjay, I've left some general comments.  Things look great overall, just a few code structure things to address.  And of course the response-format issue you mentioned.

I tried to reproduce that but actually had some trouble even invoking the APIs without running into errors.  If not, I can work through the errors I'm seeing, but if you had a few `curl` commands or whatever you could share, it'd save me lots of time.

**Comment by iamsanjay** [comment]
> Hi @iamsanjay , sorry for the delay in getting to this. I've been mostly attending to a blocker for the JAX-RS and v2 API work, [SOLR-16531](https://issues.apache.org/jira/browse/SOLR-16531). But with that recently cleared away I'm turning back to these API conversions. Hopefully the delay hasn't put you off of collaborating here!
> 
> Going to test your change out locally today and hopefully we can understand why the response format doesn't look quite right.

I am glad that things back on track. I will go through your comments and then work on the changes you have suggested. Thank you so much for getting back on this.

File: solr/core/src/test/org/apache/solr/handler/admin/api/BackupCoreAPIJerseyTest.java [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+import org.apache.solr.request.SolrQueryRequestBase;
+import org.apache.solr.response.SolrQueryResponse;
+import org.glassfish.hk2.api.Factory;
+import org.glassfish.hk2.utilities.binding.AbstractBinder;
+import org.glassfish.jersey.process.internal.RequestScoped;
+import org.glassfish.jersey.server.ResourceConfig;
+import org.glassfish.jersey.test.JerseyTest;
+import org.junit.BeforeClass;
+
+public class BackupCoreAPIJerseyTest extends JerseyTest {
```
[the remark this answers is not in the document]
Comment (iamsanjay) [reply solr/core/src/test/org/apache/solr/handler/admin/api/BackupCoreAPIJerseyTest.java:47]: For some reason (I don't remember now) I was not able to create backup using JerseyTest as there were some initialization issues. So I created two test: A traditional way to test the back up process, and the Jersey Test to test Jersey endpoints. I will run again the JerseyTest again and share the error here.

**Pull request closed without merging**